tools/jdbc/src/main/java/org/duckdb/DuckDBTimestamp.java

```diff
@@ -41,6 +47,10 @@ public static LocalDateTime toLocalDateTime(long timeMicros) {
 				, nanosPartMicros(timeMicros), ZoneOffset.UTC);
 	}
 
+	public static OffsetDateTime toOffsetDateTime(long timeMicros) {
+		return OffsetDateTime.of(toLocalDateTime(timeMicros), ZoneOffset.UTC);
```

Comment:
One thing I'm not sure how to deal with here is that there may be a `TimeZone` setting in the `DBConfig`. If it is defined, should it be used instead of UTC here?

Reply:
Though the result is the correct instant and can be changed easily to any time zone, I can see that it might be better to return a certain offset.
How the actual setting of TimeZone can be retrieved within a query? The only really compatible format is probably the utc_offset.

Reply:
Hmm the setting is [in the extension code](https://github.com/duckdb/duckdb/blob/271ccd2bdefd05f0e3f9a4fa0d551a69f4379d16/extension/icu/icu-extension.cpp#L329) so do you have access to the db instance or the db config?  But if you need the offset instead of the TZ that could be unfortunate as the offset depends on the instant.

Reply:
The DBConfig is available via: 
```
auto ctx = stmt_ref->stmt->context;                                                                                  
  1     auto &db_config = DBConfig::GetConfig(*ctx); 
```
How do I get from there to the timezone?

I would certainly recommend that the offset is always calculated at the same place. I guess it must be the db, as SQL functions might be used. In general neither duckdb nor the client can be really sure what offset has to be used for an instant. Relaying on the DB is speculative for the user of the result.

Reply:
After a bit of reading [1], it looks like Postgresql is always returning the OffsetDateTime with a ZoneOffset.UTC. 

Considering this and the simple way to change the ZoneOffset later on (for the end-user), I would leave this method as it is now.

[1] https://jdbc.postgresql.org/documentation/head/java8-date-time.html